sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredential.java

```diff
@@ -42,14 +42,30 @@
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
         return Mono.fromSupplier(() -> {
-            if (configuration.contains(Configuration.PROPERTY_AZURE_CLIENT_ID)
-                && configuration.contains(Configuration.PROPERTY_AZURE_CLIENT_SECRET)
-                && configuration.contains(Configuration.PROPERTY_AZURE_TENANT_ID)) {
+            if (configuration.containsAll(Configuration.PROPERTY_AZURE_TENANT_ID,
```

Comment:
Thoughts on changing the `containsAll` method to `getAll`? There are already some race condition chances going on with validating existence then retrieval, if we retrieval all into local instances we could reduce the number of timing issues happening.